pkg/clients/v2/factory_test.go

```diff
@@ -406,19 +390,7 @@ func TestClientCache_createAutoScalingClient(t *testing.T) {
 	options := getOptions[autoscaling.Client, autoscaling.Options](client)
 	require.NotNil(t, options)
 
-	t.Run("Does not enable FIPS", func(t *testing.T) {
-		assert.Equal(t, options.EndpointOptions.UseFIPSEndpoint, aws.FIPSEndpointStateUnset)
-	})
-
-	t.Run("Can resolve govcloud urls", func(t *testing.T) {
-		endpoint, err := options.EndpointResolver.ResolveEndpoint("us-gov-east-1", options.EndpointOptions)
-		assert.NoError(t, err)
-		assert.Equal(t, "https://autoscaling.us-gov-east-1.amazonaws.com", endpoint.URL)
-
-		endpoint, err = options.EndpointResolver.ResolveEndpoint("us-gov-west-1", options.EndpointOptions)
-		assert.NoError(t, err)
-		assert.Equal(t, "https://autoscaling.us-gov-west-1.amazonaws.com", endpoint.URL)
-	})
```

Comment:
This is no longer easily testable since endpoint resolution is deferred until call time. EndpointResolveV2 resolve function has a different signature `ResolveEndpoint(ctx context.Context, params EndpointParameters)` where `EndpointParameters` explicitly controls FIPS/other params. These are all mapped as a part of each API call vs the old signature.